panic: unhandled error: notify.FSEventsMustScanSubDirs|notify.FSEventsKernelDropped

#### System information
 
~ geth version
Geth
Version: 1.4.10-stable-5f55d95a
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.6.2
OS: darwin
GOPATH=
GOROOT=/usr/local/Cellar/go/1.6.2/libexec

#### Expected behaviour


#### Actual behaviour

Software crash

#### Steps to reproduce the behaviour
No steps found to reproduce

#### Backtrace

````
I0717 23:50:36.332543 core/blockchain.go:963] imported 1 block(s) (0 queued 0 ignored) including 0 txs in 17.456699ms. #1903995 [5b577140 / 5b577140]
panic: unhandled error: notify.FSEventsMustScanSubDirs|notify.FSEventsKernelDropped

goroutine 37 [running]:
panic(0x48f1740, 0xc82a92f560)
	/usr/local/opt/go/libexec/src/runtime/panic.go:481 +0x3e6
github.com/rjeczalik/notify.(*watch).Dispatch(0xc820177700, 0xc82f032500, 0x1, 0x1)
	/private/tmp/ethereum-20160716-84588-1tmln98/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go:136 +0x54f
github.com/rjeczalik/notify.(*watch).Dispatch-fm(0xc82f032500, 0x1, 0x1)
	/private/tmp/ethereum-20160716-84588-1tmln98/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go:205 +0x3e
github.com/rjeczalik/notify.gostream(0x5e032d0, 0x1, 0x1, 0x7007550, 0x7007560, 0x7007570)
	/private/tmp/ethereum-20160716-84588-1tmln98/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:104 +0x11c
github.com/rjeczalik/notify._cgoexpwrap_b138320e5461_gostream(0x5e032d0, 0x1, 0x1, 0x7007550, 0x7007560, 0x7007570)
	??:0 +0x53
github.com/rjeczalik/notify._Cfunc_CFRunLoopRun()
	??:0 +0x36
github.com/rjeczalik/notify.init.1.func1()
	/private/tmp/ethereum-20160716-84588-1tmln98/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:68 +0x112
created by github.com/rjeczalik/notify.init.1
	/private/tmp/ethereum-20160716-84588-1tmln98/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:70 +0x48
````

Probably the same error here. Nothing relevant running in parallel.

Steps to reproduce: (i) delete Library/Ethereum/chaindata, (ii) start geth (rescans blockchain)

$ ./build/bin/geth version
Geth
Version: 1.5.0-unstable-a4c4125b
Protocol Versions: [63 62]
Network Id: 1
Go Version: go1.6.2
OS: darwin
GOPATH=
GOROOT=/usr/local/Cellar/go/1.6.2/libexec


```
I0802 10:47:24.007685 core/blockchain.go:971] imported 2048 block(s) (0 queued 0 ignored) including 2486 txs in 4.858215456s. #412053 [90c0310c / ce00b78b]
panic: unhandled error: notify.FSEventsKernelDropped|notify.FSEventsMustScanSubDirs

goroutine 22 [running]:
panic(0x4906d60, 0xc82eb0e600)
        /usr/local/Cellar/go/1.6.2/libexec/src/runtime/panic.go:481 +0x3e6
github.com/rjeczalik/notify.(*watch).Dispatch(0xc820196000, 0xc825b12e60, 0x1, 0x1)
        /Users/ar/Code/Ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go:136 +0x54f
github.com/rjeczalik/notify.(*watch).Dispatch-fm(0xc825b12e60, 0x1, 0x1)
        /Users/ar/Code/Ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go:205 +0x3e
github.com/rjeczalik/notify.gostream(0x8500100, 0x1, 0x1, 0x5f02a70, 0x5f06440, 0x5f02ac0)
        /Users/ar/Code/Ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:104 +0x11c
github.com/rjeczalik/notify._cgoexpwrap_b138320e5461_gostream(0x8500100, 0x1, 0x1, 0x5f02a70, 0x5f06440, 0x5f02ac0)
        ??:0 +0x53
github.com/rjeczalik/notify._Cfunc_CFRunLoopRun()
        ??:0 +0x36
github.com/rjeczalik/notify.init.1.func1()
        /Users/ar/Code/Ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:68 +0x112
created by github.com/rjeczalik/notify.init.1
        /Users/ar/Code/Ethereum/go-ethereum/build/_workspace/src/github.com/ethereum/go-ethereum/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents_cgo.go:70 +0x48

```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -112,7 +112,7 @@
 		},
 		{
 			"ImportPath": "github.com/rjeczalik/notify",
-			"Rev": "5dd6205716539662f8f14ab513552b41eab69d5d"
+			"Rev": "f627deca7a510d96f0ef9388f2d0e8b16d21f87f"
 		},
 		{
 			"ImportPath": "github.com/robertkrimen/otto",

--- a/Godeps/_workspace/src/github.com/rjeczalik/notify/.travis.yml
+++ b/Godeps/_workspace/src/github.com/rjeczalik/notify/.travis.yml
@@ -21,10 +21,9 @@ env:
    - PATH=$HOME/bin:$PATH
 
 install:
- - go get golang.org/x/tools/cmd/vet
  - go get -t -v ./...
 
 script:
- - go tool vet -all .
+ - "(go version | grep -q 1.4) || go tool vet -all ."
  - go install $GOFLAGS ./...
  - go test -v -race $GOFLAGS ./...

--- a/Godeps/_workspace/src/github.com/rjeczalik/notify/appveyor.yml
+++ b/Godeps/_workspace/src/github.com/rjeczalik/notify/appveyor.yml
@@ -11,7 +11,6 @@ environment:
 
 install:
  - go version
- - go get golang.org/x/tools/cmd/vet
  - go get -v -t ./...
 
 build_script:

--- a/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go
+++ b/Godeps/_workspace/src/github.com/rjeczalik/notify/watcher_fsevents.go
@@ -133,7 +133,7 @@ func (w *watch) Dispatch(ev []FSEvent) {
 			ev[i].Flags, ev[i].Path, i, ev[i].ID, len(ev))
 		if ev[i].Flags&failure != 0 {
 			// TODO(rjeczalik): missing error handling
-			panic("unhandled error: " + Event(ev[i].Flags).String())
+			continue
 		}
 		if !strings.HasPrefix(ev[i].Path, w.path) {
 			continue

